Implemented Devise for Organization Owner Registration

MehakNadeem opened the pull request:
Implement Devise for Owner Registration

Added Devise configuration and views for user authentication (registrations, sessions, passwords, etc.)
Created custom mailers for Devise notifications
Updated stylesheets from CSS to SCSS with new organization and user-specific styles
Added controllers and views for organizations and user management
Included asset images

umairahmed786 committed:
Install the cancan gem and define abilities  for different roles

MehakNadeem committed:
Implemented Devise for Organization Owner Registration

MehakNadeem committed:
Implemented Devise for Organization Owner Registration

MehakNadeem committed:
Implemented Devise for Organization Owner Registration

dawoodmalhi commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,65 @@
+/*
```
add app/assets/stylesheets/application.css in .gitignore

dawoodmalhi commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,65 @@
+   border-radius: 5px;
+ }
+ 
+ .flash.notice {
+   background-color: #e7f3fe;
+   color: #31708f;
+ }
```
```suggestion
 .flash {
   .notice {
     background-color: #e7f3fe;
     color: #31708f;
   }
 }
```

dawoodmalhi commented on app/assets/stylesheets/application.scss:
```diff
@@ -0,0 +1,65 @@
+   color: #a94442;
+ }
+ 
+ #error_explanation {
+   border: 1px solid #f5c6cb;
+   background-color: #f8d7da;
+   padding: 20px;
+   border-radius: 5px;
+   margin-bottom: 20px;
+ }
+ 
+ #error_explanation h2 {
+   color: #721c24;
+   font-size: 18px;
+   margin-top: 0;
+ }
+ 
+ #error_explanation ul {
+   margin: 0;
+   padding: 0;
+   list-style-type: none;
+ }
+ 
+ #error_explanation ul li {
+   color: #721c24;
+   font-size: 14px;
+ }
```
use scss conventions here.

dawoodmalhi commented on app/controllers/organizations_controller.rb:
```diff
@@ -0,0 +1,35 @@
+class OrganizationsController < ApplicationController
+  skip_before_action :set_tenant, only: %i[new create index]
```
```suggestion
  skip_before_action :set_tenant
```

dawoodmalhi commented on app/controllers/organizations_controller.rb:
```diff
@@ -0,0 +1,35 @@
+    owner_role = Role.find_by(name: 'owner')
+    if @organization.save
+      subdomain = @organization.subdomain
+      host_with_subdomain = "#{subdomain}.localhost"
```
create a constant for domain and use it here instead of localhost

dawoodmalhi commented on app/controllers/organizations_controller.rb:
```diff
@@ -0,0 +1,35 @@
+
+  def create
+    @organization = Organization.new(organization_params)
+    owner_role = Role.find_by(name: 'owner')
```
```suggestion
    owner_role_id = Role.find_by(name: 'owner').id
```

dawoodmalhi commented on app/controllers/organizations_controller.rb:
```diff
@@ -0,0 +1,35 @@
+        host: host_with_subdomain,
+        port:,
+        path: new_user_registration_path,
+        query: { role_id: owner_role.id }.to_query
```
```suggestion
        query: { role_id: owner_role_id }.to_query
```

dawoodmalhi commented on app/helpers/organizations_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module OrganizationsHelper
```
remove this file.

dawoodmalhi commented on app/mailers/user_mailer.rb:
```diff
@@ -0,0 +1,40 @@
+  end
+
+  def reset_password_instructions(record, token, opts = {})
+    opts[:subject] = "Reset Password Instructions"
```
```suggestion
    opts[:subject] = 'Reset Password Instructions'
```

dawoodmalhi commented on app/views/devise/confirmations/new.html.erb:
```diff
@@ -0,0 +1,16 @@
+<h2>Resend confirmation instructions</h2>
```
use internationalization everywhere in the code.

dawoodmalhi commented on test/controllers/organizations_controller_test.rb:
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
if not using remove this file.

dawoodmalhi commented on package-lock.json:
```diff
@@ -0,0 +1,12790 @@
+{
```
do we need this file in this PR?

umairahmed786 committed:
Implement small changes recommended by mentor.

umairahmed786 committed:
Specify the version in the gemfile and change spelling of the role.

MehakNadeem committed:
Merge branch 'development' into feature/users_handling_with_devise

itratzahra7 commented on app/views/users/confirmations/new.html.erb:
```diff
@@ -0,0 +1,16 @@
+<h2>Resend confirmation instructions</h2>
```
internationalisation

itratzahra7 commented on Gemfile:
```diff
@@ -6,11 +6,17 @@ ruby '3.1.4'
+gem 'cancancan'
+# Use sqlite3 as the database for Active Record
+gem 'bootstrap_form', '~> 5.4'
+gem 'bootstrap-sass'
```
add versions

MehakNadeem committed:
Implemented devise feature

MehakNadeem committed:
PR comments resolved:
-> scss conventions followed
-> en.yml file updated
-> devise.en.yml file updated

MehakNadeem committed:
Node module files deleted

MehakNadeem committed:
Devise deleted files

MehakNadeem committed:
node modules files deleted

MehakNadeem committed:
removed unnecessary changes

MehakNadeem committed:
deleted package-lock.json file from this branch

MehakNadeem committed:
deleted unneccessary helper and test files

MehakNadeem committed:
Deleted Automated Devise Controllers

MehakNadeem committed:
Merge branch 'development' of github.com:umairahmed786/ticketing-task into feature/users_handling_with_devise

MehakNadeem committed:
Gem versions specified
